client.go

```diff
@@ -95,7 +95,7 @@ func NewClient[Req, Res any](httpClient HTTPClient, url string, options ...Clien
 		response, err := receiveUnaryResponse[Res](conn, config.Initializer)
 		if err != nil {
 			_ = conn.CloseResponse()
-			return nil, err
+			return response, err
```

Comment:
I guess I could argue that every returning a nil, it should be a properly typed nil?

So maybe something like:

```go
var zero *Res
return zero, err
```

or

```go
return (*Res)(nil), err
```

Reply:
If you'd like to guarantee that the response is empty it would be:
```go
return NewResponse[Res](nil), err
```
or more simply:
```go
return &Response[Res]{}, err
```

Which is fine by me, not here to argue.

Reply:
Alternatively, [this err != nil check](https://github.com/connectrpc/connect-go/blob/74a6754f29185b85fefa2915bf8fb680a36ca8f0/client.go#L113) could be updated:

```go
if err != nil || response == nil {
	return nil, err
}
```
Let me know which way you'd like to go -- this might be the preferred approach because it keeps back-compat: there's no change to what interceptors receive for an error response.